tools/layers.py

```diff
@@ -143,3 +146,22 @@ def evaluate_expressions(
     if exp.hasEvalError():
         raise QgsPluginExpressionException(bar_msg=bar_msg(exp.evalErrorString()))
     return value
+
+
+def get_field_index(layer: QgsVectorLayer, field_name: str) -> int:
+    """
+    Get field index if exists
+    :param layer: QgsVectorLayer
+    :param field_name: name of the field
+    :return: index of the field
+    """
+    field_index = layer.fields().indexFromName(field_name)
+    if field_index == -1:
+        raise QgsPluginLayerException(
```

Comment:
Could we raise just a standard KeyError instead of a custom one?

Reply:
Good idea!